halo2_gadgets/src/poseidon/primitives/mds.rs

```diff
@@ -82,7 +82,12 @@ pub(super) fn generate_mds<F: FromUniformBytes<64> + Ord, const T: usize>(
                 acc
             } else {
                 // We can invert freely; by construction, the elements of xs are distinct.
-                acc * (x - x_m) * (x_j - x_m).invert().unwrap()
+                
+                // We hard-code the type, to avoid wasm-pack compiler error of: cannot infer type. 
+                let denominator: F = x_j - x_m;
+                denominator.invert().unwrap();
+                
+                acc * (x - x_m) * denominator
```

Comment:
This fails tests because [`Field::invert`](https://docs.rs/ff/0.13.0/ff/trait.Field.html#tymethod.invert) is not a mutating method, so the result of `denominator.invert()` is being dropped. Instead, you need to use the output of `denominator.invert()` in the original equation:
```suggestion
                // We hard-code the type, to avoid wasm-pack compiler error of: cannot infer type. 
                let denominator: F = x_j - x_m;
                acc * (x - x_m) * denominator.invert().unwrap()
```

Reply:
Thx will fix asap